tsdb/db.go

```diff
@@ -617,6 +610,14 @@ func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs
 		return nil, err
 	}
 
+	// Register metrics after assigning the head block else it causes panics is the gauge function.
```

Comment:
Grammar is off here, I'd end the sentence at block.

Reply:
It should be `in the gauge function`. I will cut the sentence short.